### DISCUSSION TITLE
How to do deployment of Strapi using Jenkins

### DISCUSSION THREAD

**Discussion by Prateekmaithil19** [strapi/strapi#7395] [Legacy Topics]
Every time  getting the below error
![Error backend](https://user-images.githubusercontent.com/69189733/89752465-b74c4a00-dafe-11ea-8a84-bca33355c9e9.PNG)

**Comment by derrickmehaffy** [comment]
Hello can you please confirm two things:

- What node version is jenkins trying to use here?
- Are you installing the node modules before building?

**Reply by Prateekmaithil19** [reply]
Thank you for the response.

I am using NodeJs 14.7.0 Version in Jenkins
No, I am pulling the development code from Git & in Jenkins i used the command npm install
npm run build
For reference my package.json is like below:
{
"name": "planworx-backend",
"private": true,
"version": "0.1.0",
"description": "A Planworx Microsoft Event Application",
"scripts": {
"develop": "strapi develop",
"start": "strapi start",
"build": "strapi build",
"strapi": "strapi",
"pretty": "prettier --write '**/*.js'"
},
"devDependencies": {},
"prettier": {
"singleQuote": true,
"useTabs": true,
"semi": false,
"quoteProps": "as-needed",
"trailingComma": "none"
},
"dependencies": {
"knex": "<0.20.0",
"mysql": "^2.18.1",
"nodemailer": "^6.4.10",
"prettier": "^2.0.5",
"socket.io": "^2.3.0",
"strapi": "3.0.1",
"strapi-admin": "3.0.1",
"strapi-connector-bookshelf": "3.0.1",
"strapi-plugin-content-manager": "3.0.1",
"strapi-plugin-content-type-builder": "3.0.1",
"strapi-plugin-documentation": "3.0.1",
"strapi-plugin-email": "3.0.1",
"strapi-plugin-upload": "3.0.1",
"strapi-plugin-users-permissions": "3.0.1",
"strapi-utils": "3.0.1"
},
"author": {
"name": "Dev"
},
"strapi": {
"uuid": "dfa72a71-0303-4c89-96f6-61c7644c9fa3"
},
"engines": {
"node": ">=10.0.0",
"npm": ">=6.0.0"
},
"license": "MIT

**Reply by derrickmehaffy** [reply]
Yes in this case the issue is with Node v14 which isn't supported yet. Strapi only supports Node LTS versions (v12 currently, though it will work with v10) and Node v14 doesn't go LTS until october. You will most generally hit issues with the `sharp` package for building the admin as it also doesn't support Node v14 yet.

**Reply by Prateekmaithil19** [reply]
Now, I used the Nodejs 10.0 version in jenkins & tried to build now getting below error:

> planworx-backend@0.1.0 build /var/lib/jenkins/workspace/gitlabbackend
> strapi build

Error while running command build: file.isFile is not a function
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! planworx-backend@0.1.0 build: `strapi build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the planworx-backend@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/lib/jenkins/.npm/_logs/2020-08-10T18_34_45_296Z-debug.log
Build step 'Shell ausführen' marked build as failure
Finished: FAILURE

**Reply by derrickmehaffy** [reply]
Not sure I understand where this error is coming from, can you provide any more context?

**Reply by Prateekmaithil19** [reply]
Sure. 
My app frontend is in NodeJs and back-end is using Strapi.

I want to deploy the app both front-end as well back-end on Linux machine using Jenkins.

While I am running npm build command for my front-end project. I got the structure like this:
![Build Structure](https://user-images.githubusercontent.com/69189733/89885985-18a31480-dbf6-11ea-8084-4ff74b33746f.PNG)

And when I am trying to build back-end, I am getting this error
![backend failure](https://user-images.githubusercontent.com/69189733/89886231-7b94ab80-dbf6-11ea-9bb4-aa55061b9e74.PNG)

I used the command like
 npm install sharp
npm install 
npm run build


Remark: My Jenkins is hosted on one server

**Reply by derrickmehaffy** [reply]
That is odd :thinking: I'm not sure where you are getting that issue from `file.isFile`.

As a foot note you don't need to install sharp manually, that depends is included with the `strapi-plugin-upload` and newer versions of the package may not work correctly.
